runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -4885,18 +4944,25 @@ void amHandleAMO(struct amRequest_AMO_t* amo) {
   doCpuAMO(amo->obj, &amo->opnd, &amo->cmpr, &result,
            amo->ofiOp, amo->ofiType, amo->size);
 
+  bool putDone = true;
   if (amo->result != NULL) {
-    CHK_TRUE(mrGetKey(NULL, NULL, amo->b.node, amo->result, resSize));
-    (void) ofi_put(&result, amo->b.node, amo->result, resSize);
+    if (true) {
```

Comment:
I suspect this is just temporary logic, but I'd simply re-test `amo->result` below on the if-stmt that decides whether to do an `amPutDone()` rather than relying on a proxy flag. Or even do the debug `am_reqDoneStr()` debug output and then either `amrequestFMOResult()` or `amPutDone()`, as appropriate.

Reply:
This was just temporary until we figured out the logic we wanted. I'll be replacing it with (some also temporary) environment variables for performance testing.